app/controllers/application_controller/buttons.rb

```diff
@@ -147,7 +177,58 @@ def automate_button_field_changed
         page.replace("form_role_visibility", :partial => "layouts/role_visibility", :locals => {:rec_id => (@custom_button.id || "new").to_s, :action => "automate_button_field_changed"})
       end
       unless params[:target_class]
-        @changed = session[:changed] = (@edit[:new] != @edit[:current])
+        @changed = session[:changed] = false
+
+        # This block checks if any of the fields have changed
+        # Broken into different branches for specific fields
+        # If no fields are changed it will not set @changed or session[:changed] to true
+        @edit[:new].each_key do |key|
+          if @edit[:new][key] != @edit[:current][key]
+            if @edit[:new][key] == "" && @edit[:current][key].nil? # check empty string / nil case
+              next
+            elsif @edit[:new][key] == @edit[:current][key].to_s # check string / integer case
+              next
+            elsif key == :roles # check role values
+              if (@edit[:new][key] == ["_ALL_"] || @edit[:new][key] == []) && (@edit[:current][key].nil? || @edit[:current][key] == ["all"] || @edit[:current][key] == ["_ALL_"]) # if visibility is set to "To All"
```

Comment:
This conditional is pretty complicated.  I read it as `if (new is all or none) and (current is all or none)`.  Is it possible to normalize this more so we don't have so many ways to specify all or none?  Maybe in in a followup PR?